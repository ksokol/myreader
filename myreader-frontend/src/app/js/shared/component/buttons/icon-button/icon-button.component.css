my-icon-button {
    background: transparent;
}

.my-icon-button {
    background-color: inherit;
    border-radius: 50%;
    border: 1px;
    cursor: pointer;
    height: 30px;
    margin: 10px;
    padding: 0;
    user-select: none;
    width: 30px;
    -webkit-tap-highlight-color: transparent;
}

@media (min-width: 1280px) {
    .my-icon-button {
        height: inherit;
        width: inherit;
    }
}
