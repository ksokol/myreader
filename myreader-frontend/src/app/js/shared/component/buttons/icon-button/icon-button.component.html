<button type="button" class="my-icon-button">
    <react-component name="Icon" props="$ctrl.props"></react-component>
</button>
