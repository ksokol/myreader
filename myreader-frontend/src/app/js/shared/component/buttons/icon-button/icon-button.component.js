import template from './icon-button.component.html'
import './icon-button.component.css'

export const IconButton = {
    template,
    bindings: {
        myType: '@',
        myColor: '@'
    }
}
