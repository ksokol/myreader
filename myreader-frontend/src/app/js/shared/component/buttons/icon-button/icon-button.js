import React from 'react'
import PropTypes from 'prop-types'
import {Icon} from '../../icon/icon'

export const IconButton = props => <button type="button" className="my-icon-button"><Icon {...props}></Icon></button>

Icon.propTypes = {
    type: PropTypes.string.isRequired,
    color: PropTypes.string
}
