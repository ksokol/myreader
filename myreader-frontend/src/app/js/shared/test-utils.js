import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {initialApplicationState} from '../store'
import ShallowRenderer from 'react-test-renderer/shallow'

/*
 * https://velesin.io/2016/08/23/unit-testing-angular-1-5-components/
 */
export function componentMock(name) {
  function _componentMock($provide) {
    _componentMock.bindings = {}

    $provide.decorator(name + 'Directive', function ($delegate) {
      const component = $delegate[0]
      component.template = '<ng-transclude></ng-transclude>'
      component.transclude = true
      component.controller = function () {
        _componentMock.bindings = this
      }

      return $delegate
    })
  }

  return _componentMock
}

export function reactComponent(name) {
  function _reactComponentMock($provide) {
    $provide.value(name, props => {
      _reactComponentMock.bindings = {...props}
      return ''
    })
  }

  return _reactComponentMock
}

export function multipleComponentMock(name) {
  multipleComponentMock.bindings = []

  function _componentMock($provide) {
    _componentMock.bindings = multipleComponentMock.bindings

    $provide.decorator(name + 'Directive', function ($delegate) {
      const component = $delegate[0]
      component.template = '<ng-transclude></ng-transclude>'
      component.transclude = true
      component.controller = function () {
        multipleComponentMock.bindings.push(this)
      }

      return $delegate
    })
  }

  return _componentMock
}

export function filterMock(name) {
  function _filterMock($provide) {
    const filter = jest.fn(value => {
      if (typeof value === 'object') {
        // remove Angular specific attributes
        delete value.$$hashKey
        delete value.object
      }
      return name + '(' + JSON.stringify(value) + ')'
    })
    $provide.value(name + 'Filter', filter)
  }

  return _filterMock
}

export function createMockStore(middlewares = []) {
  let state = initialApplicationState()
  const store = configureMockStore([thunk, ...middlewares])(() => state)

  store.getActionTypes = () => store.getActions().map(it => it.type)
  store.setState = stateSlice => state = {...state, ...stateSlice}

  return store
}

export function ngReduxMock() {
  const store = createMockStore()
  const storeDispatch = store.dispatch
  store.dispatch = jest.fn(action => storeDispatch(action))

  const storeSetState = store.setState
  let mapDispatchToTargets = []

  const createMapToTarget = (component, mapStateToTarget) => state =>
    typeof mapStateToTarget === 'function' ? Object.assign(component, mapStateToTarget(state)) : () => null

  store.connect = jest.fn((mapStateToTarget, mapDispatchToTarget) => {
    return component => {
      const fn = createMapToTarget(component, mapStateToTarget)
      fn(store.getState())
      mapDispatchToTargets.push(fn)

      if (typeof mapDispatchToTarget === 'function') {
        Object.assign(component, mapDispatchToTarget(store.dispatch))
      }
      return () => {
      }
    }
  })

  store.setState = stateSlice => {
    storeSetState(stateSlice)
    mapDispatchToTargets.forEach(mapDispatchToTarget => mapDispatchToTarget(store.getState()))
  }

  return store
}

export function mockNgRedux() {
  function _mock($provide) {
    $provide.value('$ngRedux', ngReduxMock())
  }

  return _mock
}

export function onKey(type, event, funcs = {}) {
  let keyEvent = document.createEvent('Event')
  keyEvent.keyCode = event.keyCode
  keyEvent.key = event.key
  Object.assign(keyEvent, funcs)

  keyEvent.initEvent(`key${type}`)
  document.dispatchEvent(keyEvent)
}

export function dispatchKeyEventOnElement(element, {type, event, funcs = {}}) {
  let keyEvent = document.createEvent('Event')
  keyEvent.keyCode = event.keyCode
  keyEvent.key = event.key
  Object.assign(keyEvent, funcs)

  keyEvent.initEvent(`key${type}`)
  element.dispatchEvent(keyEvent)
}

export function tick(millis = 0) {
  jest.advanceTimersByTime(millis)
}

/**
 * @deprecated Use {@link shallow(Component)} instead.
 */
export function shallowInstance(Component) {
  const renderer = new ShallowRenderer()
  renderer.render(Component)
  return renderer
}

/**
 * @deprecated Use {@link shallow(Component)} instead.
 */
export function shallowOutput(Component) {
  return shallowInstance(Component).getRenderOutput()
}

export function shallow(Component) {
  const renderer = new ShallowRenderer()
  renderer.render(Component)
  return {
    output: () => renderer.getRenderOutput()
  }
}
