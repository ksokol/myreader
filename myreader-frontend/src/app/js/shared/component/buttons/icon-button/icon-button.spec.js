import React from 'react'
import {shallow} from '../../../test-utils'
import {IconButton} from './icon-button'

describe('src/app/js/shared/component/buttons/icon-button/icon-button.spec.js', () => {

    it('should render icon button with close icon and white color', () => {
        expect(shallow(<IconButton type="close" color="white" />)).toMatchSnapshot()
    })
})
