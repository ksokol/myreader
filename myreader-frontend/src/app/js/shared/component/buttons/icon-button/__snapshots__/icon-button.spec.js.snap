// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/js/shared/component/buttons/icon-button/icon-button.spec.js should render icon button with close icon and white color 1`] = `
<button
  className="my-icon-button"
  type="button"
>
  <Icon
    color="white"
    type="close"
  />
</button>
`;
